src/Microsoft.AspNetCore.Hosting/Internal/HostingEventSource.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.Tracing;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Hosting.Internal
+{
+    [EventSource(Name = "Microsoft-AspNetCore-Hosting")]
+    public sealed class HostingEventSource : EventSource
+    {
+        public static readonly HostingEventSource Log = new HostingEventSource();
+
+        private HostingEventSource() { }
+
+        [Event(1, Level = EventLevel.Informational)]
+        public void HostStart()
+        {
+            WriteEvent(1);
+        }
+
+        [Event(2, Level = EventLevel.Informational)]
+        public void HostStop()
+        {
+            WriteEvent(2);
+        }
+
+        [NonEvent]
+        public void RequestStart(HttpContext context)
+        {
+            if (IsEnabled())
+            {
+                RequestStart(
+                    context.TraceIdentifier,
+                    context.Request.Protocol,
+                    context.Request.Method,
+                    context.Request.ContentType ?? string.Empty,
+                    context.Request.ContentLength.HasValue ? context.Request.ContentLength.Value.ToString() : string.Empty,
+                    context.Request.Scheme,
+                    context.Request.Host.ToString(),
+                    context.Request.PathBase,
+                    context.Request.Path,
+                    context.Request.QueryString.ToString());
+            }
+        }
+
+        [NonEvent]
+        public void RequestStop(HttpContext context, Exception exception = null)
+        {
+            if (IsEnabled())
+            {
+                RequestStop(
```

Comment:
Another thing I'd like to see in RequestStop is the RemoteIp. I think this would then cover most of the essential fields required to do IIS-like request logging.

It's important that the IP is in RequestStop and not just in RequestStart because if the application runs behind a proxy and uses the `ForwardedHeadersMiddleware`, RequestStart will always use the proxy IP.

However, if this mechanism should be used for request logging, the question will arise how people can log custom fields which won't be possible with this. But people could always write/use different middleware so I think this should be fine?!